### PR TITLE
Lower delkfutt lua

### PR DESCRIPTION
**Pull request by wrenffxi**

### CODE REVIEW

File: scripts/zones/Lower_Delkfutts_Tower/npcs/_540.lua [hunk 1]
```diff
@@ -8,13 +8,12 @@
...
 require("scripts/globals/missions")
+require("scripts/globals/npc_util")
 -----------------------------------
 
 function onTrade(player, npc, trade)
-    if player:getCurrentMission(WINDURST) == A_NEW_JOURNEY and player:getVar("MissionStatus") == 2 then
-        if trade:hasItemQty(549, 1) and trade:getItemCount() == 1 then -- Delkfutt Key
-            player:startEvent(2)
-        end
+    if player:getCurrentMission(WINDURST) == A_NEW_JOURNEY and player:getVar("MissionStatus") == 2 and npcUtil.tradeHas(trade, 549) then -- Delkfutt Key
```
Comment (TeoTwawki) [inline scripts/zones/Lower_Delkfutts_Tower/npcs/_540.lua]: wide line..

File: scripts/zones/Lower_Delkfutts_Tower/npcs/_541.lua [hunk 2]
```diff
@@ -8,13 +8,12 @@
...
 require("scripts/globals/missions")
+require("scripts/globals/npc_util")
 -----------------------------------
 
 function onTrade(player, npc, trade)
-    if player:getCurrentMission(SANDORIA) == APPOINTMENT_TO_JEUNO and player:getVar("MissionStatus") == 4 then
-        if trade:hasItemQty(549, 1) and trade:getItemCount() == 1 then -- Delkfutt Key
-            player:startEvent(0)
-        end
+    if player:getCurrentMission(SANDORIA) == APPOINTMENT_TO_JEUNO and player:getVar("MissionStatus") == 4 and npcUtil.tradeHas(trade, 549) then -- Delkfutt Key
```
Comment (TeoTwawki) [inline scripts/zones/Lower_Delkfutts_Tower/npcs/_541.lua]: 152 chars no likey

File: scripts/zones/Lower_Delkfutts_Tower/npcs/_542.lua [hunk 3]
```diff
@@ -8,13 +8,12 @@
...
 require("scripts/globals/missions")
+require("scripts/globals/npc_util")
 -----------------------------------
 
 function onTrade(player, npc, trade)
-    if player:getCurrentMission(BASTOK) == JEUNO_MISSION and player:getVar("MissionStatus") == 2 then
-        if trade:hasItemQty(549, 1) and trade:getItemCount() == 1 then -- Delkfutt Key
-            player:startEvent(1)
-        end
+    if player:getCurrentMission(BASTOK) == JEUNO_MISSION and player:getVar("MissionStatus") == 2 and npcUtil.tradeHas(trade, 549) then -- Delkfutt Key
```
Comment (TeoTwawki) [inline scripts/zones/Lower_Delkfutts_Tower/npcs/_542.lua]: ..

File: scripts/zones/Lower_Delkfutts_Tower/npcs/_545.lua [hunk 4]
```diff
@@ -16,9 +16,9 @@ function onTrigger(player, npc)
         player:getCurrentMission(COP) == THREE_PATHS and
         player:getVar("COP_Tenzen_s_Path") == 6 and
         player:hasKeyItem(dsp.ki.DELKFUTT_RECOGNITION_DEVICE) and
-        not GetMobByID(ID.mob.DISASTER_IDOL):isSpawned()
+        npcUtil.popFromQM(player, npc, ID.mob.DISASTER_IDOL, {hide = 0})
     then
-        SpawnMob(ID.mob.DISASTER_IDOL):updateClaim(player)
+        -- no further action
     elseif player:getCurrentMission(COP) == THREE_PATHS and player:getVar("COP_Tenzen_s_Path") == 7 and player:hasKeyItem(dsp.ki.DELKFUTT_RECOGNITION_DEVICE) then
```
Comment (TeoTwawki) [inline scripts/zones/Lower_Delkfutts_Tower/npcs/_545.lua]: not this pr's doing but this one too is wider than style guide

File: scripts/zones/Lower_Delkfutts_Tower/npcs/_544.lua [hunk 5]
```diff
@@ -4,44 +4,35 @@
...
+function onTrigger(player, npc)
+    if player:hasKeyItem(dsp.ki.DELKFUTT_KEY) then
+        player:startEvent(16)
     else
-        player:startEvent(10); -- door is firmly shut
+        player:startEvent(10) -- door is firmly shut
     end
 
-    return 1;
+    return 1
```
Comment (TeoTwawki) [inline scripts/zones/Lower_Delkfutts_Tower/npcs/_544.lua]: while we're in lower delkfutt scripts would you mind nuking this return in each script and making sure that doesn't cause an issue? the onTrigger only has these 2 possibilities and both of them result in an event call, so the door -shouldn't- need the return to keep it from popping open on the client.

Comment (TeoTwawki) [reply scripts/zones/Lower_Delkfutts_Tower/npcs/_544.lua]: if its a bother I can merge as is and come back to it later

Comment (wrenffxi) [reply scripts/zones/Lower_Delkfutts_Tower/npcs/_544.lua]: will check it this evening

Comment (wrenffxi) [reply scripts/zones/Lower_Delkfutts_Tower/npcs/_544.lua]: tested. the returns do nothing, whether they return 1 or 0. removed them all.

**Pull request merged**